tensorflow/python/util/nest.py

```diff
@@ -142,7 +152,21 @@ def _sequence_like(instance, args):
         d[key] = result[key]
       return d
     else:
-      return instance_type((key, result[key]) for key in instance)
+      d = instance_type()
+      for key in instance:
+        d[key] = instance[key]
+      return d
+  elif _is_mapping(instance):
+    result = dict(zip(_sorted(instance), args))
+    instance_type = type(instance)
+    tf_logging.log_first_n(
+      tf_logging.WARN, "Mapping types may not work well with tf.nest. Prefer using" 
+      "MutableMapping for {}".format(instance_type), 1
+    )
+    d = instance_type()
```

Comment:
Here, we'd have to use the old constructor (because the mapping is immutable, can't be modified once constructed): `return instance_type((key, result[key]) for key in instance)`